tests/unittests/audio/test_dnsmos.py

```diff
@@ -0,0 +1,257 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from functools import partial
+from typing import Any, Dict, Optional
+
+import numpy as np
+import pytest
+import torch
+from torch import Tensor
+from torchmetrics.audio.dnsmos import DeepNoiseSuppressionMeanOpinionScore
+from torchmetrics.functional.audio.dnsmos import (
+    DNSMOS_DIR,
+    _load_session,
+    deep_noise_suppression_mean_opinion_score,
+)
+from torchmetrics.utilities.imports import (
+    _LIBROSA_AVAILABLE,
+    _ONNXRUNTIME_AVAILABLE,
+    _REQUESTS_AVAILABLE,
+)
+
+from unittests._helpers import seed_all
+from unittests._helpers.testers import MetricTester
+
+if _LIBROSA_AVAILABLE and _ONNXRUNTIME_AVAILABLE and _REQUESTS_AVAILABLE:
+    import librosa
+    import onnxruntime as ort
+else:
+    librosa, ort = None, None  # type:ignore
+
+    class InferenceSession:  # type:ignore
+        """Dummy InferenceSession."""
+
+        def __init__(self, **kwargs: Dict[str, Any]) -> None: ...
+
+
+SAMPLING_RATE = 16000
+INPUT_LENGTH = 9.01
+
+
+class _ComputeScore:
+    """The implementation from DNS-Challenge."""
+
+    def __init__(self, primary_model_path, p808_model_path) -> None:
+        self.onnx_sess = ort.InferenceSession(os.path.expanduser(primary_model_path))
+        self.p808_onnx_sess = ort.InferenceSession(os.path.expanduser(p808_model_path))
+
+    def _audio_melspec(self, audio, n_mels=120, frame_size=320, hop_length=160, sr=16000, to_db=True):
+        mel_spec = librosa.feature.melspectrogram(
+            y=audio, sr=sr, n_fft=frame_size + 1, hop_length=hop_length, n_mels=n_mels
+        )
+        if to_db:
+            mel_spec = (librosa.power_to_db(mel_spec, ref=np.max) + 40) / 40
+        return mel_spec.T
+
+    def _get_polyfit_val(self, sig, bak, ovr, is_personalized):
+        if is_personalized:
+            p_ovr = np.poly1d([-0.00533021, 0.005101, 1.18058466, -0.11236046])
+            p_sig = np.poly1d([-0.01019296, 0.02751166, 1.19576786, -0.24348726])
+            p_bak = np.poly1d([-0.04976499, 0.44276479, -0.1644611, 0.96883132])
+        else:
+            p_ovr = np.poly1d([-0.06766283, 1.11546468, 0.04602535])
+            p_sig = np.poly1d([-0.08397278, 1.22083953, 0.0052439])
+            p_bak = np.poly1d([-0.13166888, 1.60915514, -0.39604546])
```

Comment:
This seems like a near duplicate with function `_polyfit_val` in `src/torchmetrics/functional/audio/dnsmos.py`

Would be cleaner if that function was used here.